internal/pkg/csumio/csumreader_test.go

```diff
@@ -0,0 +1,194 @@
+package csumio_test
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/cespare/xxhash/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cybozu-go/fin/internal/pkg/csumio"
+)
+
+func TestChecksumReader_ChecksumMismatch_VerificationEnabled(t *testing.T) {
+	// Description:
+	// Check that checksum verification detects checksum mismatch when verification is enabled
+	//
+	// Arrange:
+	// - Data file contains test data
+	// - Checksum file contains checksum for different data (intentionally wrong)
+	// - Checksum verification is enabled
+	//
+	// Act:
+	// Read data using ChecksumReader
+	//
+	// Assert:
+	// Should fail with ErrChecksumMismatch
+
+	chunkSize := 4096 // 4KiB
+	cases := []struct {
+		name string
+		data []byte
+	}{
+		{name: "exact_chunk", data: bytes.Repeat([]byte("a"), 4096)},        // exactly one chunk
+		{name: "partial_chunk", data: bytes.Repeat([]byte("a"), 4095)},      // less than one chunk
+		{name: "more_than_chunk", data: bytes.Repeat([]byte("a"), 4097)},    // more than one chunk
+		{name: "two_chunks_aligned", data: bytes.Repeat([]byte("a"), 8192)}, // exactly two chunks
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Arrange
+			wrongData := bytes.Repeat([]byte("b"), len(tc.data)) // Different pattern, same length
+			wrongChecksum := xxhash.Sum64(wrongData)
+			checksumBuf := make([]byte, csumio.ChecksumLen)
+			binary.LittleEndian.PutUint64(checksumBuf, wrongChecksum)
```

Comment:
We don't have to calculate xxhash to generate a wrong checksum. Writing something like `0xdeadbeef` (cf. https://ja.wikipedia.org/wiki/Hexspeak) will be enough.